ngraph_bridge/ngraph_freshness_tracker.h

```diff
@@ -114,15 +114,15 @@ class NGraphFreshnessTracker : public ResourceBase {
   void RemoveTensor(const void* base_pointer);
 
   // Removes the user function from the freshness_map_
-  void RemoveUser(std::shared_ptr<ngraph::runtime::Executable> user);
+  void RemoveUser(const std::shared_ptr<ngraph::runtime::Executable>& user);
 
  private:
   // mutex protecting the freshness_map_
   mutex mu_;
   // for each base pointer (of tensor), maintains a set of ng::Functions that
   // use it
   // Each ng function in the set is then a user of the base_pointer
-  std::map<const void*, std::set<std::shared_ptr<ngraph::runtime::Executable>>>
+  std::map<const void*, std::set<ngraph::runtime::Executable*>>
```

Comment:
We don't want to save raw pointers - as they may become null under the hood. We need to save a unique_ptr or shared_ptr. But in this case we are getting a shared_ptr from nGraph - so we are going to stay with shared_ptr.

Reply:
@avijit-nervana My concern is about performance. I see the use count of ngraph::Executable goes up to hundreds. And the increment/decrement is implemented with atomic operation and therefore the overhead is quit significant.
If we are sure that the lifespan of executable is longer than freshness tracker, it is still worth the change. You are definitely more familiar with that and can make the decision.

Reply:
@XWangIntel I agree with you on this. I personally do not like shared ptrs just for this reason. We will do this clean up a few weeks later - so will track this definitely.